### ISSUE TITLE
Extract maintain `shell` API externally

### ISSUE THREAD

**Issue by ysndr** [rust-lang/cargo#6079]
Hej there

I recently extracted the [`shell` API](https://github.com/rust-lang/cargo/blob/master/src/cargo/core/shell.rs) into its own repo at [ysndr/shell-rs](https://github.com/ysndr/shell-rs) for easier usage in my own projects.

I wonder if it's desired to maintain cargo's shell API externally as well in this repo as part of the rust-lang-nursery to offer the advantages of a multiplatform, colored, customizable `stdout`wrapper to the whoel community.

Please excuse if this is the wrong place to ask for this I just did'nt know a better place.

**Comment by alexcrichton** [comment]
I think this would be a great crate to split out! I'd prefer it live in this repository so we can make changes if necessary, but we could certainly publish it all and take prs for docs and such

**Comment by ysndr** [comment]
I think as the project is strongly coupled with cargo by nature including it in the repo is certainly a good idea. Im just worried that new features might get ignored because of the lack of relevance in cargo which would be a point for putting it into it's own repo, to loosen the ties to cargo a bit.

**Comment by alexcrichton** [comment]
True! Perhaps the API could be extracted in this repo, and if it goes well after a few weeks/months/etc we can extract it to its own repo?

**Comment by ysndr** [comment]
Yes maybe its best to build a stable foundation in this repo. 
Yet I think it should get its own repo as soon as it's considered kinda final (API-wise and usage wise inside cargo). Otherwise having to keep track with cargo just to contribute to shell could keep people from doing so.
It could also be considered too close to cargo to be used alone.

tldr; I think you right, stabilizing it's API usage and documentation/testing(?) should be done here be its main users. Let's give it some more room for itself when that is done

**Comment by epage** [comment]
Closing in favor of #5088
